iphone/Maps/Classes/ClipboardMonitor.mm

```diff
@@ -0,0 +1,107 @@
+#import "MapsAppDelegate.h"
+#import "ClipboardMonitor.h"
+#import "UIKitCategories.h"
+#import "MapsAppDelegate.h"
+#import <CoreApi/DeepLinkParser.h>
+#include <CoreApi/Framework.h>
+#import <CoreApi/DeepLinkData.h>
+#import "UIKitCategories.h"
+#import <CoreApi/DeepLinkSearchData.h>
+
+@interface ClipboardWatcher ()
+
+@property (nonatomic, strong) NSTimer *timer;
+@property (nonatomic, copy) NSString *previousClipboardText;
+@property (nonatomic, assign) BOOL isErrorPromptShown;
+
+@end
+
+@implementation ClipboardWatcher
+
+- (void)startWatching:(UIWindow *)window {
+    NSLog(@"clip watching started");
+
+    UIPasteboard *pasteboard = [UIPasteboard generalPasteboard];
+    NSString *currentClipboardText = pasteboard.string;
+    NSLog(@"clip clipboard checking for changes %@  and  %@",currentClipboardText, self.previousClipboardText);
+
+    [pasteboard setString:@""];
+    [self checkClipboardWithText:currentClipboardText window:window];
+}
+
+- (void)checkClipboardWithText:(NSString *)copiedText window:(UIWindow *)window {
+    NSLog(@"clip API CALL PROCEEDING %@", copiedText);
+
+    // Check if the copied text is a URL
+    if ([self isURL:copiedText]) {
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Redirecting from copied URL" message:@"Please wait..." preferredStyle:UIAlertControllerStyleAlert];
```

Comment:
1. Text should be different, something like "Extracting coordinates from %@url using the anonymous proxy..."
2. All text translations are stored in data/strings/strings.txt and regenerated using tools/unix/generate_localizations.sh in a separate commit (there is a doc about it).

Reply:
Changed it.